src/alexandria3k/data_sources/uspto.py

```diff
@@ -633,7 +646,19 @@ class Uspto(DataSource):
         data files are located
     :type uspto_directory: str
 
-    # TODO: Add sampling.
+    :param sample: A callable that returns a tuple of two values
+        to control Zip file and container sampling. It defaults
+        to `lambda x: (True, True)`. The population or query method
+        will call this argument for each USPTO Zip file path as
+        its argument and for each container info as its argument.
+        When the first callable element of the tuple returns `True`
```

Comment:
There are not two callable elements.  It's just one callable returning a tuple.

Please also clarify how the callable can know whether the passed data is a zip file or container info.  Perhaps the arguments should also be a tuple?

Reply:
In Python 3.* versions it is not allowed to pass a tuple as an argument of a lambda function, see [here](https://peps.python.org/pep-3113/). 

Following the lead of the PEP, I can change the sample variable to be:

`sample=lambda x: (x[0], x[1])`, 

which works and I think looks nicer. Let me know your opinion on this and I will change it.

Also, I need some help to understand why the callable should know what the passed data is. 

After giving the argument to the callable I explicitly ask a value of the two-value tuple it returns, and w.r.t. that sampling will happen or not. 

Perhaps, I should clarify the last step on the initialization of the callable and the calls of it.
This takes place on the __init__() of the DataFiles() in uspto.py and on the uspto_zip_cache.py, so I think I should add clarifications on these parts as well.

Reply:
I added the changes and pushed it in the form I am describing on my previous comment, so you can see what I am saying.